src/posix/platform/spi_interface.cpp

```diff
@@ -830,7 +835,12 @@ otError SpiInterface::SendFrame(const uint8_t *aFrame, uint16_t aLength)
     mSpiTxIsReady     = true;
     mSpiTxPayloadSize = aLength;
 
-    IgnoreError(PushPullSpi());
+    while (count < kMaxNumRetries)
+    {
+        VerifyOrExit((error = PushPullSpi()) != OT_ERROR_NONE);
+        usleep(1000);
+        count++;
+    }
```

Comment:
I am not sure of the intention behind this change:

- We are basically calling `PushPull()` 5 times in row each time adding 1msec delay.
- This is then adding 5msec delay in main run loop of OT stack for every SPI transaction which is not good.
- Once we call `PushPull()`, if we have any pending tx frame we sent to peripheral and if it indicates that it accepted it, we clear tx frame. 
- Then the follow up 4 calls of `PushPull()` will simply send empty transactions.
- If the peripheral could not accept the frame, then we would anyway retry the same tx frame in next time we get to `Process()` and do `PushPull()`.

Reply:
I think we quit immediately if PushPullSpi is successfully. So, in most cases, we don't retry at all, hence no delay.